src/main/java/org/fenixedu/academic/service/services/manager/MergeExecutionCourses.java

```diff
@@ -424,6 +426,14 @@ private static void copyPersistentGroups(ExecutionCourse executionCourseFrom, Ex
         }
     }
 
+    private static void copySenderMessages(ExecutionCourse executionCourseFrom, ExecutionCourse executionCourseTo) {
+        if (executionCourseFrom.getSender() != null) {
+            ExecutionCourseSender courseSenderTo = ExecutionCourseSender.newInstance(executionCourseTo);
+            courseSenderTo.getMessagesSet().addAll(executionCourseFrom.getSender().getMessagesSet());
+            executionCourseFrom.getSender().delete();
```

Comment:
From the code in the Recipient's `delete` method, any message associated with it, will be deleted.

As this does not dissociate those messages from the recipients, they will be deleted.